datafusion/optimizer/src/eliminate_cross_join.rs

```diff
@@ -402,7 +400,18 @@ fn remove_join_expressions(expr: Expr, join_keys: &JoinKeySet) -> Option<Expr> {
                 _ => None,
             }
         }
-
+        Expr::BinaryExpr(BinaryExpr { left, op, right }) if op == Operator::Or => {
+            let l = remove_join_expressions(*left, join_keys);
+            let r = remove_join_expressions(*right, join_keys);
+            match (l, r) {
+                (Some(ll), Some(rr)) => Some(Expr::BinaryExpr(BinaryExpr::new(
+                    Box::new(ll),
+                    op,
+                    Box::new(rr),
+                ))),
+                _ => None,
```

Comment:
When either `left` or `right` is empty, it means they are `true`.

Reply:
That might be useful to encode as a comment:
```suggestion
               // When either `left` or `right` is empty, it means they are `true`
               // so OR'ing anything with them will also be true
                _ => None,
```